pkg/controller/compute/router.go

```diff
@@ -0,0 +1,181 @@
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type routerConnector struct {
+	kube client.Client
+}
+
+func (c *routerConnector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	projectID, opts, err := gcp.GetAuthInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, err
+	}
+	s, err := compute.NewService(ctx, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+	return &routerExternal{Service: s, kube: c.kube, projectID: projectID}, nil
+}
+
+type routerExternal struct {
+	kube client.Client
+	*compute.Service
+	projectID string
+}
+
+func (c *routerExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1beta1.Router)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotRouter)
+	}
+	observed, err := c.Routers.Get(c.projectID, cr.Spec.ForProvider.Region, meta.GetExternalName(cr)).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(gcp.IsErrorNotFound, err), errGetRouter)
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	router.LateInitializeSpec(&cr.Spec.ForProvider, *observed)
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errManagedRouterUpdate)
+		}
+	}
+
+	cr.Status.AtProvider = router.GenerateRouterObservation(*observed)
+
+	cr.Status.SetConditions(xpv1.Available())
+
+	u, err := router.IsUpToDate(meta.GetExternalName(cr), &cr.Spec.ForProvider, observed)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckRouterUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: u,
+	}, nil
+}
+
+func (c *routerExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1beta1.Router)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotRouter)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
+
+	rt := &compute.Router{}
+	router.GenerateRouter(meta.GetExternalName(cr), cr.Spec.ForProvider, rt)
+	_, err := c.Routers.Insert(c.projectID, cr.Spec.ForProvider.Region, rt).
+		Context(ctx).
+		Do()
+	return managed.ExternalCreation{}, errors.Wrap(err, errRouterCreateFailed)
+}
+
+func (c *routerExternal) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1beta1.Router)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotRouter)
+	}
+
+	observed, err := c.Routers.Get(c.projectID, cr.Spec.ForProvider.Region, meta.GetExternalName(cr)).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(gcp.IsErrorNotFound, err), errGetRouter)
```

Comment:
```suggestion
		return managed.ExternalUpdate{}, errors.Wrap(err, errGetRouter)
```
I believe we should not ignore not found error at this point, given we are in `Update`